### PR TITLE
support config k8s resource server

### PR DESCRIPTION
**Pull request by huiwq1990**
/kind feature
> /kind question
> /kind design
> /sig ai
> /sig iot
> /sig network
> /sig storage
> /sig storage

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by Congrool** [comment]
@huiwq1990 Thanks for your contribution. And could you please add some details about why we need this option?

**Comment by huiwq1990** [comment]
> @huiwq1990 Thanks for your contribution. And could you please add some details about why we need this option?

In private delivery scenes, network maybe not allow access public domain , it's better allow custom define resource address.

**Comment by rambohe-ch** [comment]
> > @huiwq1990 Thanks for your contribution. And could you please add some details about why we need this option?
> 
> In private delivery scenes, network maybe not allow access public domain , it's better allow custom define resource address.

@huiwq1990 Thanks for your feedback. it looks reasonable to add this parameter, @Congrool how about your advices?

File: pkg/yurtctl/cmd/join/join.go [hunk 1]
```diff
@@ -159,21 +161,26 @@ func addJoinConfigFlags(flagSet *flag.FlagSet, joinOptions *joinOptions) {
 		&joinOptions.nodeLabels, options.NodeLabels, joinOptions.nodeLabels,
 		"Sets the labels for joining node",
 	)
+	flagSet.StringVar(
+		&joinOptions.kubernetesResourceServer, options.KubernetesResourceServer, joinOptions.kubernetesResourceServer,
+		"Sets the address for download k8s node resources",
```
Comment (Congrool) [inline pkg/yurtctl/cmd/join/join.go:166]: Maybe we should modify it as `Sets the address for downloading k8s node resources`

**Review by Congrool — COMMENTED** [review]
It looks good :-), please correct the garmar.

File: pkg/yurtctl/cmd/join/join.go (hunk 1 above)
Comment (huiwq1990) [reply pkg/yurtctl/cmd/join/join.go]: Just correct it.

**Comment by rambohe-ch** [comment]
@huiwq1990 Thanks for fixing comments, please merge two commits into one commit.

**Comment by huiwq1990** [comment]
> @huiwq1990 Thanks for fixing comments, please merge two commits into one commit.

Already done.

**Review by Congrool — COMMENTED** [review]
/lgtm

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**